File: backend/internal/model/Room.go
package model

import "gorm.io/gorm"

type Room struct {
	gorm.Model
	Code         string   `json:"code"`
	Name         string   `json:"name"`
	Capacity     string   `json:"capacity"`
	RoomTypeID   uint     `json:"room_type_id"`
	RoomType     RoomType `gorm:"foreignKey:RoomTypeID"`
	Floor        int      `json:"floor"`
	Building     string   `json:"building"`
	Observations string   `json:"observations"`
}

type RoomType struct {
	gorm.Model
	Name string `json:"name"`
}

File: algoritm-service/src/models/room.rs
use genevo::{operator::prelude::*, prelude::*, random::Rng, types::fmt::Display};
use serde::{Deserialize, Serialize};
use std::fmt;

// ==============================
// Estructura para cursos
// ==============================
#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
pub struct Room {
    #[serde(rename = "ID")]
    pub id: usize,
    pub code: String,
    pub name: String,
    pub capacity: u32,
    pub room_type: usize,
    pub floor: u32,
    pub building: String,
    pub observations: String,
}
